Source/Controllers/Other/SecureBookmarkData.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  SecureBookmarkData.swift
+//  Sequel Ace
+//
+//  Created by James on 7/12/2020.
+//  Copyright © 2020 Sequel-Ace. All rights reserved.
+//
+
+import Foundation
+
+final class SecureBookmarkData: NSObject {
+    let bookmarkData: Data
+    private let options: Double
+    private let bookmarkURL: URL
+
+    init(data: Data, options: Double, url: URL) {
+        bookmarkData = data
+        self.options = options
+        bookmarkURL = url
+        super.init()
+    }
+}
+
+extension SecureBookmarkData: NSCoding, NSSecureCoding {
+    static var supportsSecureCoding: Bool {
+        return true
+    }
+
+    // MARK: NSCoding Implementation
+
+    enum Keys: String {
+        case bookmarkData = "BookmarkData"
+        case options = "Options"
+        case bookmarkURL = "BookmarkURL"
+    }
+
+    func encode(with coder: NSCoder) {
+        if #available(OSX 10.13, *) {
+            // For NSSecureCoding
+            coder.encode(bookmarkData as NSData, forKey: Keys.bookmarkData.rawValue)
+            coder.encode(NSNumber(value: options), forKey: Keys.options.rawValue)
+            coder.encode(bookmarkURL as NSURL, forKey: Keys.bookmarkURL.rawValue)
```

Comment:
What here requires 10.13? Could we not use this same block for both secure and not-secure coding?

Reply:
Secure coding requires 10.13.